### PR TITLE
fix(typings): make 'status' state slice more typesafe

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**

Make 'status' state slice more typesafe by providing proper StatusAction type

**Test plan**

Tests are not needed

### CODE REVIEW

File: packages/netlify-cms-core/src/reducers/status.ts [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+    api: StaticallyTypedRecord<{ status: boolean; statusPage: string }>;
+  }>;
+  error: Error | undefined;
+}>;
 
-const status = (state = Map(), action: StatusAction) => {
+const defaultState = fromJS({
+  isFetching: false,
+  status: {
+    auth: { status: false },
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/reducers/status.ts:17]:
```suggestion
    auth: { status: true },
```

File: packages/netlify-cms-core/src/reducers/status.ts [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+    api: { status: false, statusPage: '' },
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/reducers/status.ts:18]:
```suggestion
    api: { status: true, statusPage: '' },
```

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @smashercosmo, looks great.
I've made two minor suggestions to the default state so it will reflect an non error status.
See https://github.com/netlify/netlify-cms/blob/2abeacfaee774951924d81536dc7db7001d3af09/packages/netlify-cms-core/src/actions/status.ts#L51
Please let me know what you think

**Comment by smashercosmo** [comment]
@erezrokah Should these also be changed then https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-core/src/backend.ts#L337-L338 ?

**Comment by erezrokah** [comment]
> @erezrokah Should these also be changed then https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-core/src/backend.ts#L337-L338 ?

Good point.- I think that default value can be removed since the first iteration of the loop will overwrite it anyway. We might need to make TypeScript happy since it might report an error of returning an uninitialized value.

**Comment by smashercosmo** [comment]
TS does yell if you remove default value. So, I'll just change default status value to 'true' for consistency.

**Comment by smashercosmo** [comment]
I would also make condition inside the loop more explicit, so instead of ```if (Object.values(status).every(s => s.status === true)) {...}``` I would just ```if (status.auth.status && status.api.status) {...}```. Easier to understand what's going on.

**Comment by erezrokah** [comment]
> I would also make condition inside the loop more explicit, so instead of `if (Object.values(status).every(s => s.status === true)) {...}` I would just `if (status.auth.status && status.api.status) {...}`. Easier to understand what's going on.

I understand but we might add more indicators in the future. I could go either way, but if you do make that change please verify all backends return that structure.

**Comment by smashercosmo** [comment]
Ok, I'll leave it as is then

**Comment by smashercosmo** [comment]
done

**Review by erezrokah — APPROVED** [review]

**Pull request merged**